resources/sdk/clisdkclient/extensions/services/commandservice.go

```diff
@@ -0,0 +1,180 @@
+package services
+
+import (
+	"encoding/json"
+	"fmt"
+	"gc/config"
+	"gc/models"
+	"gc/restclient"
+	"gc/utils"
+	"log"
+	"net/http"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/spf13/cobra"
+)
+
+//CommandService holds the method signatures for all common Command object invocations
+type CommandService interface {
+	Get(uri string) (string, error)
+	List(uri string) (string, error)
+	Post(uri string, payload string) (string, error)
+	Patch(uri string, payload string) (string, error)
+	Put(uri string, payload string) (string, error)
+	Delete(uri string) (string, error)
+	DetermineAction(httpMethod string, operationId string, uri string) (string, error)
```

Comment:
@carnellj  I've added DetermineAction to this interface to clean up the API code. Also I changed Delete to also return the body so we can see the output of it.